[docs] CDC content addition for default value of snapshot option for slot (#27257)
yugabyte/yugabyte-db: docs/content/preview/api/ysql/the-sql-language/statements/streaming_create_repl_slot.md

`NOEXPORT_SNAPSHOT` will just use the snapshot for logical decoding as normal but won't do anything else with it.

{{< note title="YSQL default snapshot option" >}}

> The default snapshot option in YugabyteDB is `USE_SNAPSHOT` when the preview flag [ysql_enable_pg_export_snapshot](../../../../explore/ysql-language-features/advanced-features/snapshot-synchronization) is enabled. Without the preview flag enabled, specifying the option at the time of slot creation will not have any effect.

Reviewer: ```suggestion
YugabyteDB uses the `USE_SNAPSHOT` option as the default when the preview flag [ysql_enable_pg_export_snapshot](../../../../explore/ysql-language-features/advanced-features/snapshot-synchronization) is enabled. If you do not enable the preview flag, specifying the option during slot creation has no effect.
```
